Description: Set UsePAM=no for regress/putty-*.sh
 Currently these tests fails in the autopkgtest infrastructure due to pam_loginuid.so
 failures. These failures cannot currently be replicated locally. Workaround this
 by setting UsePAM=no for the failing tests since their functionality is not tesing
 PAM.
Author: Nick Rosbrook <enr0n@ubuntu.com>
Forwarded: no
Last-Update: 2024-07-31
--- a/regress/putty-ciphers.sh
+++ b/regress/putty-ciphers.sh
@@ -43,6 +43,8 @@
 		echo "MACs $m" >> ${OBJ}/sshd_proxy
 	fi
 
+	sed -i "s/UsePAM.*/UsePAM no/" ${OBJ}/sshd_proxy
+
 	rm -f ${COPY}
 	env HOME=$PWD ${PLINK} -load cipher_$c -batch -i ${OBJ}/putty.rsa2 \
 	    cat ${DATA} > ${COPY}
--- a/regress/putty-kex.sh
+++ b/regress/putty-kex.sh
@@ -26,6 +26,8 @@
 	cp ${OBJ}/sshd_proxy_bak ${OBJ}/sshd_proxy
 	echo "KexAlgorithms ${k}" >>${OBJ}/sshd_proxy
 
+	sed -i "s/UsePAM.*/UsePAM no/" ${OBJ}/sshd_proxy
+
 	env HOME=$PWD ${PLINK} -v -load localhost_proxy -batch -i ${OBJ}/putty.rsa2 true \
 	    2>${OBJ}/log/putty-kex-$k.log
 	if [ $? -ne 0 ]; then
--- a/regress/putty-transfer.sh
+++ b/regress/putty-transfer.sh
@@ -5,6 +5,8 @@
 
 puttysetup
 
+sed -i "s/UsePAM.*/UsePAM no/" ${OBJ}/sshd_proxy
+
 if [ "`${SSH} -Q compression`" = "none" ]; then
 	comp="0"
 else
